lib/editing/editor.dart

```diff
@@ -126,7 +126,18 @@ class Position {
 }
 
 abstract class CodeCompleter {
-  Future<List<Completion>> complete(Editor editor);
+  Future<Completions> complete(Editor editor);
+}
+
+class Completions {
+
+  List<Completion> completionList;
```

Comment:
Generally, I think we'll want to avoid putting the type in the name. So perhaps `completions` rather than `completionList `.